EncoreTickets.SDK.Tests/IntegrationTests/BasketServiceForTodayTixTests.cs

```diff
@@ -0,0 +1,605 @@
+                upsertBasketResult = service.UpsertBasket(request);
+                const int expectedReservationsCount = 1;
+
+                var basketDetails = service.GetBasketDetails(upsertBasketResult.Reference);
+
+                AssertUpsertBasketSuccess(request, basketDetails, expectedReservationsCount);
+                Assert.AreEqual(upsertBasketResult.Reference, basketDetails.Reference);
+            }
+            finally
+            {
+                service.ClearBasket(upsertBasketResult?.Reference);
+            }
+        }
+
+        [Test]
+        public void UpsertBasket_GetBasket_IfNewBasketAndReservationItemsMoreThan1_Successful()
+        {
+            var upsertBasketResult = (Basket.Models.Basket)null;
+            try
+            {
+                var referenceItem1 = configuration["Basket:TestReferences:0"];
+                var referenceItem2 = configuration["Basket:TestReferences:1"];
+                var request = CreateDefaultBasket(referenceItem1, referenceItem2);
+                upsertBasketResult = service.UpsertBasket(request);
+                const int expectedReservationsCount = 1;
+
+                var basketDetails = service.GetBasketDetails(upsertBasketResult.Reference);
+
+                AssertUpsertBasketSuccess(request, basketDetails, expectedReservationsCount);
+                Assert.AreEqual(upsertBasketResult.Reference, basketDetails.Reference);
+            }
+            finally
+            {
+                service.ClearBasket(upsertBasketResult?.Reference);
+            }
+        }
+
+        // todo: add additional tests when the basket API v2 is fully ready
+
+        [Test]
+        public void UpsertBasket_IfReservationsMoreThan1_Exception400()
+        {
+            var referenceItem1 = configuration["Basket:TestReferences:0"];
+            var referenceItem2 = configuration["Basket:TestReferences:1"];
+            var request = CreateDefaultBasket(referenceItem1);
+            var reservation2 = request.Reservations[0].CopyObject();
+            reservation2.Items[0].AggregateReference = referenceItem2;
+            request.Reservations.Add(reservation2);
+
+            var exception = Assert.Catch<ApiException>(() =>
+            {
+                var upsertBasketResult = service.UpsertBasket(request);
+                try
+                {
+                    service.ClearBasket(upsertBasketResult?.Reference);
+                }
+                catch (Exception e)
+                {
+                    // ignored
+                }
+            });
+
+            AssertApiException(exception, HttpStatusCode.BadRequest);
+        }
+
+        [Test]
+        public void UpsertBasket_IfReservationItemIsNotAvailable_Exception400()
+        {
+            var reference = configuration["Basket:TestReferences:0"];
+            var request = CreateDefaultBasket(reference);
+            var upsertBasketResult = service.UpsertBasket(request);
```

Comment:
Because we create a new basket twice with the same seat, and for the second attempt, the requested seat will not be available because it belongs to the first basket